src/oscar/apps/order/abstract_models.py

```diff
@@ -1135,3 +1137,74 @@ def description(self):
         if self.voucher_code:
             return self.voucher_code
         return self.offer_name or u""
+
+
+@python_2_unicode_compatible
+class AbstractInvoice(models.Model):
```

Comment:
What information is stored in `Invoice` that we can't just store in the `Order` itself? i.e., what is the rationale for having a separate model?

Reply:
Idea was to keep all information, necessary to render invoice in the separate model in case if order removed.